Added CustomGlobalExceptionHandler and enhance DTO validation

ShastkivRuslan opened the pull request:
Added CustomGlobalExceptionHandler for centralized exception handling and enhance DTO validation

- Implemented CustomGlobalExceptionHandler to manage application-wide exceptions and improve error responses
- Added @CoverImage annotation to validate URL format for book cover images
- Added @Isbn annotation to validate ISBN format for book ISBNs in CreateBookRequestDto

ShastkivRuslan committed:
Add CustomGlobalExceptionHandler for centralized exception handling and enhance DTO validation

- Implemented CustomGlobalExceptionHandler to manage application-wide exceptions and improve error responses
- Added @CoverImage annotation to validate URL format for book cover images
- Added @Isbn annotation to validate ISBN format for book ISBNs in CreateBookRequestDto

Elena-Bruyako commented on src/main/java/ruslan/shastkiv/bookstore/controller/BookController.java:
```diff
@@ -34,7 +35,7 @@ public BookDto getBokById(@PathVariable Long id) {
 
     @PostMapping
-    public BookDto createBook(@RequestBody CreateBookRequestDto bookDto) {
+    public BookDto createBook(@Valid @RequestBody CreateBookRequestDto bookDto) {
```
Do we need `@Valid` for other endpoints?

Elena-Bruyako commented on src/main/java/ruslan/shastkiv/bookstore/dto/CreateBookRequestDto.java:
```diff
@@ -1,14 +1,30 @@
 
 @Data
 public class CreateBookRequestDto {
+    @NotNull(message = "Title cannot be null")
```
use `@NotBlank` for String

```suggestion
    @NotBlank(message = "Title cannot be null")
```

Elena-Bruyako reviewed (changes requested):
see comments

ShastkivRuslan committed:
Changes after mentor review;
-changed annotations @NotNull to @NotBlank for String fields;
-added @Valid annotation for all methods that receive CreateBookRequestDto as input;
//mvn clean package -> build success

ShastkivRuslan committed:
Some improvements:
-enhanced Global Exception Handler to manage EntityNotFoundException with user-friendly error messages.
-removed unused DataProcessingException;

ShastkivRuslan committed:
Some improvements:
-standardized error responses in GlobalExceptionHandler to use a Map structure for consistency in error handling.
